cmd/geth/dbcmd.go

```diff
@@ -551,16 +551,8 @@ func freezerInspect(ctx *cli.Context) error {
 		return err
 	}
 	stack, _ := makeConfigNode(ctx)
-	defer stack.Close()
-
-	db := utils.MakeChainDatabase(ctx, stack, true)
-	defer db.Close()
-
-	ancient, err := db.AncientDatadir()
-	if err != nil {
-		log.Info("Failed to retrieve ancient root", "err", err)
-		return err
-	}
+	ancient := stack.ResolveAncient("chaindata", ctx.String(utils.AncientFlag.Name))
```

Comment:
Any particular reason for this change? Directly resolve ancient datadir without involving the chain DB?

Reply:
Yes, for the investigation here: https://github.com/ethereum/go-ethereum/issues/26483#issuecomment-1381633114 , I needed to do a `geth db inspect`, but I didn't actually have a leveldb -- I only had  a couple of index files. 

This whole `utils.MakeChainDatabase` assumes things to be pretty well ordered, but all we use it for eventually is to help resolve the ancientdir, via `db.AncientDir`. 

So this change has the same effect as the original code, but is more robust in case the data is not fully consistent / present.